app/controllers/papers_controller.rb

```diff
@@ -5,7 +5,10 @@ class PapersController < ApplicationController
   # GET /papers
   # GET /papers.json
   def index
-    @papers = Paper.all
+    @papers = Paper.select('papers.*,sum(cached_votes_up) as total_votes')
+    .joins(:references)
+    .group('papers.id')
+    .order(('total_votes DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
```

Comment:
Can you walk me through what each of these ActiveRecord methods / sql lines does?